modules/auto-update/auto-updater-module-interface.js

```diff
@@ -0,0 +1,23 @@
+const OTAutoUpdater = require('./implementation/ot-auto-updater');
+
+class AutoUpdaterModuleInterface {
+    constructor(config) {
+        this.config = config;
+        this.logger = config.logger;
+    }
+
+    initialize() {
+        this.implementation = new OTAutoUpdater(this.config);
+        return this.implementation.initialize(this.logger);
```

Comment:
I was not clear enough for previous comment, Initialize logger for otautoUpdater in it's contructor not through initialize method. So when creating an object new OTAutoUpdater(this.logger) not in initialize method